scripts/check_no_box_dyn_error.py

```
@@ -0,0 +1,211 @@
+    # method/type signatures as public.
+    pub_trait_stack: list[int] = []
+    # Generic scope stack tracking any trait/impl so a pub-trait's enter
+    # depth is popped when the whole block closes.
+    depth = 0
+    pub_trait_entry_depths: list[int] = []
+
+    accum: list[str] = []
+    accum_start_line = 0
+    accum_is_public = False
+
+    def flush(reason: str) -> None:  # reason is purely for debugging
```

Ruff (0.15.10)

[warning] Unused function argument: `reason`

(ARG001)